### PR TITLE
Process xacro file test

### LINKED ISSUE #92: Adds tests for verifying andino_description urdf parsing.
## Desired behavior
Adds some smoke tests in `andino_description` package for processing the xacro files and checking that there is no error when doing so.

### PR DESCRIPTION
**Pull request by JesusSilvaUtrera**
# 🎉 New feature

Closes #92 

## Summary
Added a pytest unit test to process the main xacro file (`andino.urdf.xacro`) to check it is well formatted.

## Test it
Build the package and then you can test it running `colcon test --packages-select andino_description --event-handler=console_direct+ --pytest-args -k test_xacro_processing`. You will see in the therminal the output of the test and if there is any error.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [x] Added example and/or tutorial
- [ ] Updated documentation (as needed)

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by JesusSilvaUtrera** [comment]
The only doubt I have here is whether it would be necessary to add to the documentation the test part or not.

File: andino_description/test/test_xacro_processing.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import os
+import pytest
+import xacro
+
+def test_xacro_processing():
+    # Get the file path
+    xacro_file_path = os.path.join(os.path.dirname(__file__), '..', 'urdf', 'andino.urdf.xacro')
```
Comment (francocipollone) [inline andino_description/test/test_xacro_processing.py:7]: is it possible to use here ament_index to get the package path and later on the xacro file path?


```python
from ament_index_python.packages import get_package_share_directory

get_package_share_directory("package_a")
```

**Review by francocipollone — COMMENTED** [review]
Thanks for adding test coverage on this

File: andino_description/test/test_xacro_processing.py (hunk 1 above)
Comment (JesusSilvaUtrera) [reply andino_description/test/test_xacro_processing.py:7]: Actually it is possible, I didn't think of it but it is the preferred approach, thanks Franco!

File: andino_description/package.xml [hunk 2]
```diff
@@ -9,13 +9,16 @@
...
   <buildtool_depend>ament_cmake</buildtool_depend>
+  <buildtool_depend>ament_cmake_python</buildtool_depend>
 
   <exec_depend>joint_state_publisher_gui</exec_depend>
   <exec_depend>robot_state_publisher</exec_depend>
   <exec_depend>ros2launch</exec_depend>
   <exec_depend>rviz2</exec_depend>
   <exec_depend>xacro</exec_depend>
 
+  <test_depend>ament_cmake_pytest</test_depend>
```
Comment (francocipollone) [inline andino_description/package.xml:20]: nit: Also add [ament_index_python](https://github.com/ament/ament_index/tree/rolling/ament_index_python) as test_depend

**Review by francocipollone — APPROVED** [review]
LGTM, one comment to go

File: andino_description/test/test_xacro_processing.py (hunk 1 above)
Comment (francocipollone) [reply andino_description/test/test_xacro_processing.py]: Thanks for addressing this. This way we can directly check xacro on install folder. So the installation process is also tested for free.

**Review by jballoffet — APPROVED** [review]
LGTM!

File: andino_description/CMakeLists.txt [hunk 3]
```diff
@@ -14,4 +15,21 @@ install(
...
+    get_filename_component(_test_name ${_test_path} NAME_WE)
+    ament_add_pytest_test(${_test_name} ${_test_path}
+      APPEND_ENV PYTHONPATH=${CMAKE_CURRENT_BINARY_DIR}
+      TIMEOUT 60
+      WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+    )
+  endforeach()
+endif()
+
+# ament_python_install_package(${PROJECT_NAME})
```
Comment (jballoffet) [inline andino_description/CMakeLists.txt:33]: Was this left commented out on purpose? Let's avoid leaving commented out code please.

File: andino_description/test/test_xacro_processing.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import os
```
Comment (jballoffet) [inline andino_description/test/test_xacro_processing.py:1]: Please add license notice.

File: andino_description/test/test_xacro_processing.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import pytest
+import xacro
+from ament_index_python.packages import get_package_share_directory
+
+andino_description_pkg = get_package_share_directory("andino_description")
+
+def test_xacro_processing():
+    # Get the file path
```
Comment (jballoffet) [inline andino_description/test/test_xacro_processing.py:9]: nit: punctuation.

File: andino_description/test/test_xacro_processing.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    xacro_file_path = os.path.join(andino_description_pkg, 'urdf', 'andino.urdf.xacro')
+
+    # Test xacro processing
```
Comment (jballoffet) [inline andino_description/test/test_xacro_processing.py:12]: nit: punctuation.

File: andino_description/test/test_xacro_processing.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import os
+import pytest
+import xacro
+from ament_index_python.packages import get_package_share_directory
+
+andino_description_pkg = get_package_share_directory("andino_description")
```
Comment (jballoffet) [inline andino_description/test/test_xacro_processing.py:6]: Is this variable required? Consider dropping it.

File: andino_description/test/test_xacro_processing.py [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+def test_xacro_processing():
```
Comment (jballoffet) [inline andino_description/test/test_xacro_processing.py:8]: Consider adding a docstring.

File: andino_description/test/test_xacro_processing.py (hunk 1 above)
Comment (JesusSilvaUtrera) [reply andino_description/test/test_xacro_processing.py]: Happy to help!

File: andino_description/test/test_xacro_processing.py [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import os
+import pytest
+import xacro
+from ament_index_python.packages import get_package_share_directory
+
+def test_xacro_processing():
```
Comment (jballoffet) [inline andino_description/test/test_xacro_processing.py:36]: Last comment: please consider adding a docstring for this method (https://peps.python.org/pep-0257/).

Comment (jballoffet) [reply andino_description/test/test_xacro_processing.py:36]: And please rebase branch, thank you!

**Comment by JesusSilvaUtrera** [comment]
@jballoffet all comments addressed.

File: andino_description/test/test_xacro_processing.py (hunk 9 above)
Comment (jballoffet) [reply andino_description/test/test_xacro_processing.py:36]: Thanks!

**Review by jballoffet — APPROVED** [review]

**Pull request merged**